runtime/src/debug/cpp/KDebug.cpp

```diff
@@ -188,6 +223,98 @@ RUNTIME_USED const char* Konan_DebugGetTypeName(KRef obj) {
   return CreateCStringFromString(type_info->relativeName_);
 }
 
+}  // namespace
+
+extern "C" {
+
+RUNTIME_USED RUNTIME_WEAK char* Konan_DebugBuffer() {
+  return Konan_DebugBufferImpl();
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugBufferSize() {
+  return Konan_DebugBufferSizeImpl();
+}
+
+RUNTIME_USED RUNTIME_WEAK char* Konan_DebugBufferWithObject(KRef obj) {
+  auto* impl = getImpl<char* (*)(KRef)>(obj, DO_DebugBufferWithObject);
+  if (impl == nullptr) return nullptr;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugBufferSizeWithObject(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugBufferSizeWithObject);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+// Auxilary function which can be called by developer/debugger to inspect an object.
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugObjectToUtf8Array(KRef obj, char* buffer, int32_t bufferSize) {
+  auto* impl = getImpl<int32_t (*)(KRef, char*, int32_t)>(obj, DO_DebugObjectToUtf8Array);
+  if (impl == nullptr) return 0;
+  return impl(obj, buffer, bufferSize);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugPrint(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugPrint);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugIsArray(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugIsArray);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugGetFieldCount(KRef obj) {
+  auto* impl = getImpl<int32_t (*)(KRef)>(obj, DO_DebugGetFieldCount);
+  if (impl == nullptr) return 0;
+  return impl(obj);
+}
+
+RUNTIME_USED RUNTIME_WEAK int32_t Konan_DebugGetFieldType(KRef obj, int32_t index) {
+  auto* impl = getImpl<int32_t (*)(KRef, int32_t)>(obj, DO_DebugGetFieldType);
+  if (impl == nullptr) return 0;
+  return impl(obj, index);
+}
+
+RUNTIME_USED RUNTIME_WEAK void* Konan_DebugGetFieldAddress(KRef obj, int32_t index) {
+  auto* impl = getImpl<void* (*)(KRef, int32_t)>(obj, DO_DebugGetFieldAddress);
+  if (impl == nullptr) return nullptr;
+  return impl(obj, index);
+}
+
+// Compute address of field or an array element at the index, or null, if incorrect.
+RUNTIME_USED RUNTIME_WEAK const char* Konan_DebugGetFieldName(KRef obj, int32_t index) {
+  auto* impl = getImpl<const char* (*)(KRef, int32_t)>(obj, DO_DebugGetFieldName);
+  if (impl == nullptr) return nullptr;
+  return impl(obj, index);
+}
+
+RUNTIME_USED RUNTIME_WEAK const char* Konan_DebugGetTypeName(KRef obj) {
+  auto* impl = getImpl<const char* (*)(KRef)>(obj, DO_DebugGetTypeName);
+  if (impl == nullptr) return nullptr;
+  return impl(obj);
+}
+
+// IMPORTANT: when updating, fix `debugOperationsSize` in RTTIGenerator.kt.
+RUNTIME_USED RUNTIME_WEAK
```

Comment:
Why are these attributes required?

Reply:
So that we could use them from Kotlin side and not conflict on dups.

Reply:
What do you mean by "Kotlin side"? Generated code?
There are plenty of runtime functions that aren't `RUNTIME_USED RUNTIME_WEAK` and can be properly used from generated code without conflicts.

Reply:
OK, if needed we can easily add it.